### PR TITLE
PT-2214: Add ability to configure login types

### PR DESCRIPTION
**Pull request by ksavosteev**
## Description
Added the ability to configure two main login types: with login/password and AzureAD.

ezgif-6-23ef12ac6410

Login/password is enabled by default. It can be disabled in configuration options in admin (new settings):
```JSON
    "PasswordLogin": {
        "Enabled": false
    },
```
AzureAD configuration options are provided out of the box (disabled by default):
```JSON
    "AzureAd": {
        "Enabled": true,
        "AuthenticationType": "AzureAD",
       ...
      },
```
If both login types are enabled the priority in which they are show can be controlled by the `Priority` option (lower go first):
```JSON
    "PasswordLogin": {
        "Enabled": true,
        "Priority": 0
    },
```
```JSON
    "AzureAd": {
        "Enabled": true,
        "Priority": 1
      },
```

### Jira-link:

 https://virtocommerce.atlassian.net/browse/PT-2214

### CODE REVIEW

File: src/VirtoCommerce.Platform.Web/Controllers/Api/AuthorizationController.cs [hunk 1]
```diff
@@ -74,6 +78,15 @@ public async Task<ActionResult> Exchange()
                     });
                 }
 
+                if (!_passwordLoginOptions.Enabled && !user.IsAdministrator)
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/AuthorizationController.cs:81]:
```suggestion
                if (_passwordLoginOptions.Disabled && !user.IsAdministrator)
```

File: src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs [hunk 2]
```diff
@@ -782,6 +790,33 @@ public async Task<ActionResult<UserApiKey[]>> DeleteUserApiKeys([FromQuery] stri
...
+        /// Get allowed login types
+        /// </summary>
+        [HttpGet]
+        [Route("logintypes")]
+        [AllowAnonymous]
+        public ActionResult GetLoginTypes()
+        {
+            var options = new List<LoginType>
+            {
+                new LoginType
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs:803]: Two identical items?

File: src/VirtoCommerce.Platform.Web/Model/Security/PasswordLoginOptions.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+namespace VirtoCommerce.Platform.Web.Model.Security
+{
+    public class PasswordLoginOptions
+    {
+        /// <summary>
+        /// Determines whether the user authentication by username/password is enabled.
+        /// Always enabled unless explicitly disabled by the platform options.
+        /// </summary>
+        public bool Enabled { get; set; } = true;
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Model/Security/PasswordLoginOptions.cs:9]:
```suggestion
        public bool Disabled { get; set; }
```

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]
Update documentation https://github.com/VirtoCommerce/vc-platform/blob/master/docs/user-guide/configuration-settings.md

File: src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs (hunk 2 above)
Comment (ksavosteev) [reply src/VirtoCommerce.Platform.Web/Controllers/Api/SecurityController.cs:803]: No, one item for each setting.

**Pull request merged**